src/middlewares/auth/userAuthentication.js

```diff
@@ -0,0 +1,19 @@
+const { UserRoles } = require('../../models');
+
+const userAuthentication = (...allowedRoles) => {
+  return async (req, res, next) => {
+    if (!req.session.passport) return res.boom.unauthorized();
+
+    const userRoles = await UserRoles.find({
+      name: { $in: allowedRoles }
+    }).exec();
+
+    for (const role of userRoles) {
```

Comment:
You can write in one line: if(userRoles.some(el => el._id.toString() === req.session.passport.user.role)) return next()

Reply:
Fixed.